packages/styled-components/src/sheet/GroupedTag.js

```diff
@@ -34,7 +34,7 @@ class DefaultGroupedTag implements GroupedTag {
     return index;
   }
 
-  insertRules(group: number, rules: string[]): void {
+  insertRules(group: number, rules: string | string[]): void {
```

Comment:
Are we sure that we want to make this function more polymorphic? And is this a special case? Afaik if we know that we only have one rule then `[x]` does the trick, so is this because of a change in Stylis?

Reply:
> Are we sure that we want to make this function more polymorphic? And is this a special case? Afaik if we know that we only have one rule then `[x]` does the trick, so is this because of a change in Stylis?

Yes stylis only emits a single rule now instead of an array. In my benchmarking this was more performant...

Reply:
This could depend though. Are we sure that a variable array sometimes and a string most of the time is properly benchmarked with attempts to evenly switch between the two to check for V8 performance cliffs? 😅 just making sure

Reply:
Good question, It's mostly rehydration where multirule inserts happen and we don't have any benchmarks for that.

Reply:
I guess making this work goes first, making it fast goes second. There's a lot of refactoring we could do anyway to slim thing down generally.